packages/react/src/components/Toolbar/ToolbarMenu.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import * as PropTypes from 'prop-types'
+
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+
+import {
+  createShorthandFactory,
+  commonPropTypes,
+  UIComponent,
+  childrenExist,
+  UIComponentProps,
+  ChildrenComponentProps,
+  ContentComponentProps,
+} from '../../lib'
+import { mergeComponentVariables } from '../../lib/mergeThemes'
+
+import { ComponentEventHandler, ShorthandCollection, withSafeTypeForAs } from '../../types'
+import { submenuBehavior } from '../../lib/accessibility'
+
+import ToolbarMenuDivider from './ToolbarMenuDivider'
+import ToolbarMenuItem, { ToolbarMenuItemProps } from './ToolbarMenuItem'
+
+export type ToolbarMenuItemShorthandKinds = 'divider' | 'item'
+
+export interface ToolbarMenuProps
+  extends UIComponentProps,
+    ChildrenComponentProps,
+    ContentComponentProps {
+  /** Shorthand array of props for ToolbarMenu. */
+  items?: ShorthandCollection<ToolbarMenuItemShorthandKinds>
+
+  /**
+   * Called on item click.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All item props.
+   */
+  onItemClick?: ComponentEventHandler<ToolbarMenuItemProps>
+}
+
+class ToolbarMenu extends UIComponent<ToolbarMenuProps> {
+  static displayName = 'ToolbarMenu'
+
+  static className = 'ui-toolbar__menu'
+
+  static create: Function
+
+  static propTypes = {
+    ...commonPropTypes.createCommon(),
+    items: customPropTypes.collectionShorthandWithKindProp(['divider', 'item']),
+    onItemClick: PropTypes.func,
+  }
+
+  static defaultProps = {
+    accessibility: submenuBehavior,
+    as: 'ul',
+  }
+
+  handleItemOverrides = variables => predefinedProps => ({
+    onClick: (e, itemProps) => {
+      _.invoke(predefinedProps, 'onClick', e, itemProps)
+      _.invoke(this.props, 'onItemClick', e, itemProps)
+    },
+    variables: mergeComponentVariables(variables, predefinedProps.variables),
```

Comment:
Should this be done in a factory?

And I don't remember exactly the case, but these `variables` silently override `ToolbarItemVariables` as I remember. Are we going to fix that in this PR?

Reply:
Related to variables inheritance.